File: CS50x/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    // Checks if given number is positive, if fails asks again till correct value is given. Then returns the value of cents.
    int cents;

    do
    {
        cents = get_int("How many cents the customer is owed? ");
    }
    while (cents < 1);

    return cents;
}

int calculate_quarters(int cents)
{
    // Returns a number of quarters that contains the given value of cents.
    int quarters = cents/25;
    return quarters;
}

int calculate_dimes(int cents)
{
    // Returns a number of dimes that contains the given value of cents (operates on value potentially reduced by previous function call inside main(void) to properly sum number of coins).
    int dimes = cents/10;
    return dimes;
}

int calculate_nickels(int cents)
{
    // Returns a number of nickels that contains the given value of cents (operates on value potentially reduced by previous function call inside main(void) to properly sum number of coins).
    int nickels = cents/5;
    return nickels;
}

int calculate_pennies(int cents)
{
    // Returns a number of pennies that contains the given value of cents (operates on value potentially reduced by previous function call inside main(void) to properly sum number of coins).
    int pennies = cents/1;
    return pennies;
}

File: CS50x/sentimental-cash/cash.py
from cs50 import get_float

def main():
    dollars = get_dollars()

    quarters = calculate_quarters(dollars)
    dollars = dollars - quarters * 0.25

    dimes = calculate_dimes(dollars)
    dollars = dollars - dimes * 0.10

    nickels = calculate_nickels(dollars)
    dollars = dollars - nickels * 0.05

    pennies = calculate_pennies(dollars)

    coins = quarters + dimes + nickels + pennies
    print(round(coins))

def get_dollars():
    while True:
        n = get_float("Change owed: ")
        if n > 0:
            break
        else:
            print("That's not a valid number!")
    return n


def calculate_quarters(dollars):
    quarters = round(dollars/0.25, 1)
    return int(quarters)

def calculate_dimes(dollars):
    dimes = round(dollars/0.10, 1)
    return int(dimes)

def calculate_nickels(dollars):
    nickels = round(dollars/0.05, 1)
    return int(nickels)

def calculate_pennies(dollars):
    pennies = round(dollars/0.01, 1)
    return int(pennies)


main()